### ISSUE TITLE
Add Drag and Drop events on WebviewViewProvider and Webview

### ISSUE THREAD

**Issue by AllanOricil** [microsoft/vscode#111092]
I want to transfer data between a WebviewViewProvider and a Webview using Drag and Drop events. The Drag start events are started from the WebviewViewProvider elements and the Drop events are received on the Webview elements. 

To achieve this I would like to have the following APIs

````js
webviewViewProvider.view.onDragStart();
webviewViewProvider.view.onDragEnd();


webview.view.onDrop();
webview.view.onDragOver();
webview.view.onDragLeave();
webview.view.onDragEnter();
````
![image](https://user-images.githubusercontent.com/55927613/99877626-741e6e80-2bf7-11eb-83b1-e2df0c440e6a.png)

Basically this feature request is to add all drag and drop event listeners on the view of both WebviewViewProvider and Webview.
The HTML of each view just need to know that something was dragged from somewhere and dropped on the webview.

**Comment by mjbvz** [comment]
Related  #124593

**Comment by mjbvz** [comment]
Maybe I am misunderstanding this issue but I'm pretty sure this works today using the standard html data transfer apis

Testing:

1. In the webview sample, add a draggle element and code such as:

```js
    window.addEventListener('dragstart', e => { 
        e.dataTransfer.setData('vscode/test', ':)');
    })
```

2. In the webview view sample, add webview code such as:

```js
window.addEventListener('dragover', e => {
    e.preventDefault();
});

window.addEventListener('drop', e => {
    e.preventDefault();
    console.log(e.dataTransfer.getData('vscode/test'));
});
```

3. Run both extensions
4. Try dragging an element from the webview to the webview view and see the correct string logged

Using the dom DataTransfer api is better IMO than having VS Code expose custom events. If you need your extension to do something on drag/drop, you can always post a message from the webview back to the extension


Let me know if I'm overlooking something with this feature request

**Comment by onfireprojects** [comment]
> 1. In the webview sample, add a draggle element and code to the webview such as:
> 
> window.addEventListener('dragstart', e => { 
>     e.dataTransfer.setData('vscode/test', ':)');
> })
> 2. In the webview view sample, add webview code such as:
> 
> window.addEventListener('dragover', e => {
>     e.preventDefault();
> });
> 
> window.addEventListener('drop', e => {
>     e.preventDefault();
>     console.log(e.dataTransfer.getData('vscode/test'));
> });
> 3. Run both extensions
> 4. Try dragging an element from the webview to the webview view and see the correct string logged
> 

This does not work fully for me.
The target webview does not get any of the drag-events until shift was pressed (once).
If there's any way to bypass this, that would be great, as it's not really usable for me this way, as the user is supposed to drag from a tree in the panel section to a webview in another view...

Implementation on my side is as explained, as far as i can see;
**webview-sample** got draggable items that use 
`e.dataTransfer.setData('vscode/test', 'dragged id: testing')`
and log the start of the drag in general

**webview-view-sample** got
`window.addEventListener('dragover', (e) => {
	e.preventDefault();
	if (e.dataTransfer) {
		console.log('DRAGOVER (target)', e.dataTransfer.getData('vscode/test'));
	} else {
		console.log('DRAGOVER (target; no dataTransfer');
	}
});`
+ drop eventListener which also - as all drag events- only fires after shift was pressed.

**Another issue:**
Even though both event listener use the exact same way to access the set data of dataTransfer;
`e.dataTransfer.getData('vscode/test')`
...only in the log inside the drop listener, the data is actually present, so it logs "dragged id: testing" while the dragover event does not yet seem to get ANY data and only logs the string without the dataTransfer data behind it (logging the dataTransfer item shows, that it's completely empty at that point)


As explained, this is just implemented in those example projects for now, but regarding my plan for extension this means, that right now i'm not getting the info that anything is currently being dragged inside the webview (before pressing shift) neither do i get the data of dataTransfer - even after pressing shift - before dropping the element. I really hope there's a solution or i'm just missing something.
